libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -936,29 +606,33 @@ void ArrayUtils::clearStorageLoop(Type const* _type) const
 				_context << Instruction::POP;
 				return;
 			}
-			// stack: end_pos pos
-
+			// stack: start_pos slotCount
+			// Initialize loop counter i = 0
+			_context << u256(0);
+			// stack: start_pos slotCount i
 			evmasm::AssemblyItem loopStart = _context.appendJumpToNew();
 			_context << loopStart;
-			// check for loop condition
-			_context <<
-				Instruction::DUP1 <<
-				Instruction::DUP3 <<
-				Instruction::GT <<
-				Instruction::ISZERO;
+			// check for loop condition: !(slotCount > i) = (i >= slotCount)
+			_context << Instruction::DUP1 << Instruction::DUP3 << Instruction::GT << Instruction::ISZERO;
 			evmasm::AssemblyItem zeroLoopEnd = _context.newTag();
 			_context.appendConditionalJumpTo(zeroLoopEnd);
-			// delete
+			// stack: start_pos slotCount i
+			// compute storage position: start_pos + i
+			_context << Instruction::DUP3 << Instruction::DUP2 << Instruction::ADD;
+			// stack: start_pos slotCount i (start_pos+i)
+			// delete storage slot
 			_context << u256(0);
 			StorageItem(_context, *_type).setToZero(SourceLocation(), false);
-			_context << Instruction::POP;
-			// increment
+			_context << Instruction::POP << Instruction::POP;
```

Comment:
If you don't need the reference to be kept on the stack, you can tell `setToZero()` that it can consume it:

```suggestion
			StorageItem(_context, *_type).setToZero(SourceLocation(), /* _removeReference = */ true);
			_context << Instruction::POP;
			// increment
			_context << Instruction::POP << Instruction::POP;
```

Otherwise it unnecessarily tries to protect it from modification by making a copy internally: https://github.com/argotorg/solidity/blob/aca05d4df9c6a959243d96d8be56cd1f63df8680/libsolidity/codegen/LValue.cpp#L459-L461

Reply:
Oh cool, wasn't aware of it. I will set `removeReference` to true.